tower-http/src/set_response_header.rs

```diff
@@ -0,0 +1,340 @@
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SetResponseHeaderLayer")
+            .field("header_name", &self.header_name)
+            .field("mode", &self.mode)
+            .field("make", &std::any::type_name::<M>())
+            .finish()
+    }
+}
+
+impl<M, B> SetResponseHeaderLayer<M, B> {
+    /// Create a new [`SetResponseHeaderLayer`].
+    pub fn new(header_name: HeaderName, make: M) -> Self
+    where
+        M: MakeHeaderValue<B>,
+    {
+        Self {
+            make,
+            header_name,
+            mode: InsertHeaderMode::OverrideExisting,
+            _marker: PhantomData,
+        }
+    }
+
+    /// Set which mode to use when inserting the header.
+    ///
+    /// Defaults to [`InsertHeaderMode::OverrideExisting`].
+    pub fn mode(mut self, mode: InsertHeaderMode) -> Self {
+        self.mode = mode;
+        self
+    }
+}
+
+/// The mode to use when inserting a header into a request or response.
+#[derive(Debug, Clone)]
+#[non_exhaustive]
+pub enum InsertHeaderMode {
+    /// Insert the header, overriding any previous values the header might have.
+    OverrideExisting,
+    /// Append the header and keep any previous values.
+    Append,
+    /// Insert the header only if it is not already present.
+    SkipIfPresent,
+}
+
+impl<B, S, M> Layer<S> for SetResponseHeaderLayer<M, B>
+where
+    M: MakeHeaderValue<B> + Clone,
+{
+    type Service = SetResponseHeader<S, M>;
+
+    fn layer(&self, inner: S) -> Self::Service {
+        SetResponseHeader {
+            inner,
+            header_name: self.header_name.clone(),
+            make: self.make.clone(),
+            mode: self.mode.clone(),
+        }
+    }
+}
+
+impl<M, B> Clone for SetResponseHeaderLayer<M, B>
+where
+    M: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            make: self.make.clone(),
+            header_name: self.header_name.clone(),
+            mode: self.mode.clone(),
+            _marker: PhantomData,
+        }
+    }
+}
+
+/// Middleware that sets a header on the response.
+#[derive(Clone)]
+pub struct SetResponseHeader<S, M> {
+    inner: S,
+    header_name: HeaderName,
+    make: M,
+    mode: InsertHeaderMode,
+}
+
+impl<S, M> SetResponseHeader<S, M> {
+    /// Create a new [`SetResponseHeader`].
+    pub fn new(inner: S, header_name: HeaderName, make: M) -> Self {
+        Self {
+            inner,
+            header_name,
+            make,
+            mode: InsertHeaderMode::OverrideExisting,
+        }
+    }
+
+    /// Set which mode to use when inserting the header.
+    ///
+    /// Defaults to [`InsertHeaderMode::OverrideExisting`].
+    pub fn mode(mut self, mode: InsertHeaderMode) -> Self {
+        self.mode = mode;
+        self
+    }
+}
+
+impl<S, M> fmt::Debug for SetResponseHeader<S, M>
+where
+    S: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SetResponseHeaderLayer")
```

Comment:
```suggestion
impl<S, M> fmt::Debug for SetResponseHeader<S, M>
where
    S: fmt::Debug,
{
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        f.debug_struct("SetResponseHeader")
```

Maybe a copy and paste error?

Reply:
Ah yes. Good catch!